pypeit/core/datacube.py

```diff
@@ -267,6 +270,228 @@ def dar_correction(wave_arr, coord, obstime, location, pressure, temperature, re
     return ra_diff, dec_diff
 
 
+def gaussian2D_cube(tup, amplitude, xo, yo, dxdz, dydz, sigma_x, sigma_y, theta, offset):
+    (x, y, z) = tup
+    xo = float(xo) + z*dxdz
+    yo = float(yo) + z*dydz
+    a = (np.cos(theta)**2)/(2*sigma_x**2) + (np.sin(theta)**2)/(2*sigma_y**2)
+    b = -(np.sin(2*theta))/(4*sigma_x**2) + (np.sin(2*theta))/(4*sigma_y**2)
+    c = (np.sin(theta)**2)/(2*sigma_x**2) + (np.cos(theta)**2)/(2*sigma_y**2)
+    norm = 1/(2*np.pi*np.sqrt(a*c-b*b))
+    g = offset + norm*amplitude*np.exp( - (a*((x-xo)**2) + 2*b*(x-xo)*(y-yo)
+                            + c*((y-yo)**2)))
+    return g.ravel()
+
+
+def gaussian2D(tup, amplitude, xo, yo, sigma_x, sigma_y, theta, offset):
```

Comment:
Documentation please. Inline comments are also strongly encouraged. Right now it does not look anyone else can maintain or modify this code.